library/nmcli.py

```diff
@@ -0,0 +1,735 @@
+                     120: "Failed"}
+        # select whether we dump additional debug info through syslog
+        self.syslogging=True
+
+    def execute_command(self, cmd, use_unsafe_shell=False, data=None):
+        if self.syslogging:
+            syslog.openlog('ansible-%s' % os.path.basename(__file__))
+            syslog.syslog(syslog.LOG_NOTICE, 'Command %s' % '|'.join(cmd))
+
+        return self.module.run_command(cmd, use_unsafe_shell=use_unsafe_shell, data=data)
+
+    def list_connection_info(self):
+
+        bus=dbus.SystemBus()
+        connection_list=[]
+
+        # Get a proxy for the base NetworkManager object
+        proxy=bus.get_object("org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager")
+        manager=dbus.Interface(proxy, "org.freedesktop.NetworkManager")
+
+        # Get all devices known to NM and list interface names
+        devices=manager.GetDevices()
+        for d in devices:
+            dev_proxy=bus.get_object("org.freedesktop.NetworkManager", d)
+            prop_iface=dbus.Interface(dev_proxy, "org.freedesktop.DBus.Properties")
+            props=prop_iface.GetAll("org.freedesktop.NetworkManager.Device")
+            connection_list.append(props['Interface'])
+            # try:
+            #     devtype = devtypes[props['DeviceType']]
+            # except KeyError:
+            #     devtype = "Unknown"
+            # print "Type: %s" % devtype
+            #
+            # print "Driver: %s" % props['Driver']
+            #
+            # try:
+            #     state = states[props['State']]
+            # except KeyError:
+            #     state = "Unknown"
+            # print "State: %s" % state
+        return connection_list
+
+    def connection_exists(self):
+        # we are going to use name and type in this instance to find if that connection exists and is of type x
+        connections=self.list_connection_info()
+
+        for con_item in connections:
+            if self.cname==con_item:
+                return True
+        return False
+
+    def connection_down(self):
+        cmd=[self.module.get_bin_path('nmcli', True)]
+        # if self.connection_exists():
+        cmd.append('con')
+        cmd.append('down')
+        cmd.append(self.cname)
+        return self.execute_command(cmd)
+
+    def create_connection_team(self):
+        cmd=[self.module.get_bin_path('nmcli', True)]
+        # if connection_exists()
+        #     self.module.fail_json(msg="this connection exists already")
+        # else:
+        # format for creating team interface
+        cmd.append('connection')
+        cmd.append('add')
+        cmd.append('type')
+        cmd.append(self.type)
+        cmd.append('con-name')
+        if self.cname is not None:
+            cmd.append(self.cname)
+        elif self.ifname is not None:
+            cmd.append(self.ifname)
+        else:
+            self.module.fail_json(msg="You haven't specified a name for the connection")
```

Comment:
Handle this case long before it gets to this method call. This method shouldn't need to access module at all.